clients/python/dd_image_client.py

```diff
@@ -0,0 +1,316 @@
+
+        response = self._make_feature_extraction_call(image_data)
+
+        if response is not None:
+            print("feature vector for " + image_file_path + " is:")
+            print(response['vals'])
+            return response['vals']
+        else:
+            print ("failed to extract features for " + image_file_path)
+            return None
+
+    def extract_features_from_url(self, url):
+        url_input_list = [url]
+        response = self._make_feature_extraction_call(url_input_list)
+
+        if response is not None:
+            print("vector for " + response['uri'] + " is:")
+            print(response['vals'])
+            return response['vals']
+        else:
+            print("error extracting features for " + url)
+            return None
+
+    def extract_features_from_url_file(self, data_file, target_file):
+        print("processing " + data_file + '\n')
+
+        urls = []
+        url_file = open(data_file, 'r')
+        for url in url_file:
+            urls.append(url.rstrip('\n'))
+
+        print("total number of image URL's to process " + str(len(urls)))
+
+        uri_to_features = {}
+        image_processed = 0
+        num_of_erros = 0
+
+        start_time = time.time()
+        while len(urls) > 0:
+            print("remaining images to process: " + str(len(urls)))
+            if self.batch_size > len(urls):
+                self.batch_size = len(urls)
+
+            urls_to_process = []
+            for i in range(0, self.batch_size):
+                urls_to_process.append(urls.pop())
+
+            t0 = time.time()
+            responses = self._make_feature_extraction_call(urls_to_process)
+
+            if responses is not None:
+                for response in responses:
+                    uri_to_features[response['uri']] = response['vals']
+                else:
+                    num_of_erros += 1
+
+            t1 = time.time()
+            image_processed = image_processed + self.batch_size
+            print("total time it took to process " + str(i) + " images was " + str(t1 - t0) + ' seconds')
+
+        full_file = open(target_file, 'w')
+        json.dump(uri_to_features, full_file)
+        full_file.close()
+
+        end_time = time.time()
+        total_time = (end_time - start_time) / 60
+
+        self._report_stats(image_processed, num_of_erros, target_file, total_time)
+
+        return uri_to_features
+
+    def extract_features_from_files_in_dir(self, dir_path, target_file_path):
+
+        image_files = self._fetch_files_from_dir(dir_path)
+        image_vectors = {}
+
+        while len(image_files) > 0:
+            img_path = image_files.pop()
+            img_data = self._pre_process_image(img_path)
+            response = self._make_feature_extraction_call(img_data)
+            image_vectors[img_path] = response['vals']
+            print("files left to process " + str(len(image_files)))
+
+        target_file = open(target_file_path, 'w')
```

Comment:
Could be useful to be a version of this that returns extracted features into memory.